exporters/prometheus/src/exporter_utils.cc

```diff
@@ -505,7 +518,7 @@ std::string PrometheusExporterUtils::MapToPrometheusName(
   }
 
   // Special case - counter
-  if (prometheus_type == prometheus_client::MetricType::Counter)
+  if (prometheus_type == prometheus_client::MetricType::Counter && !without_type_suffix)
```

Comment:
Likewise, instead of changing each if statement, use:

```
if (! without_type_suffix)
{
  ... code for Special case - counter ...
  ... code for Special case - gauge ...
}
```